### PR TITLE
Add Row Details documentation.

### PR DESCRIPTION
**Pull request by mariaterzieva12**

### CODE REVIEW

**Comment by mariaterzieva12** [comment]
Need to add code snippets from SDK.

File: controls/datagrid/theming-and-styles/columns-styling.md [hunk 1]
```diff
@@ -110,6 +110,20 @@ And this is how the column style looks when the properties for customizing the c
 
 ![DataGrid Column Footer](../images/column-footer-style.png)
 
+## DataGrid Toggle Row Details Column Styling
+
+`CellContentStyle` defines the appearance of the `DataGridToggleRowDetailsColumn`. The `DataGridToggleRowDetailsCellStyle` exposes the following properties for styling:
+
+* `ButtonFontFamily`&mdash;Defines the font family for the toggle symbol.
+* `ExpandButtonText`&mdash;Defines the text for the expand state.
```
Comment (yordan-mitev) [inline controls/datagrid/theming-and-styles/columns-styling.md:118]:
```suggestion
* `ExpandButtonText`&mdash;Defines the text for the expanded state.
```

File: controls/datagrid/theming-and-styles/columns-styling.md [hunk 2]
```diff
@@ -110,6 +110,20 @@ And this is how the column style looks when the properties for customizing the c
...
+* `CollapseButtonText`&mdash;Defines the text for the collapse state.
```
Comment (yordan-mitev) [inline controls/datagrid/theming-and-styles/columns-styling.md:119]:
```suggestion
* `CollapseButtonText`&mdash;Defines the text for the collapsed state.
```

File: controls/datagrid/theming-and-styles/columns-styling.md [hunk 3]
```diff
@@ -110,6 +110,20 @@ And this is how the column style looks when the properties for customizing the c
...
+* `ButtonTextColor`&mdash;Defines the color for the toggle symbol.
+* `SelectedButtonTextColor`&mdash;Defines the color for the toggle symbol when the item is selected.
+* `HoverButtonTextColor`&mdash;Defines the color for the toggle symbol when the item is hovered.
```
Comment (yordan-mitev) [inline controls/datagrid/theming-and-styles/columns-styling.md:122]:
```suggestion
* `HoverButtonTextColor`&mdash;Defines the color for the toggle symbol when the item is being hovered over.
```

File: controls/datagrid/theming-and-styles/columns-styling.md [hunk 4]
```diff
@@ -110,6 +110,20 @@ And this is how the column style looks when the properties for customizing the c
...
+* `ButtonFontSize`&mdash;Defines font size for the toggle symbol.
```
Comment (yordan-mitev) [inline controls/datagrid/theming-and-styles/columns-styling.md:123]:
```suggestion
* `ButtonFontSize`&mdash;Defines the font size for the toggle symbol.
```

File: controls/datagrid/row-details/row-details-template.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+---
+title: Row Details Template
+page_title: .NET MAUI DataGrid Documentation - Row Details Template
+description: Learn how to present additional information about DataGrid row by using row details functionality.
```
Comment (yordan-mitev) [inline controls/datagrid/row-details/row-details-template.md:4]: That's a good description!
```suggestion
description: Learn how to present additional information about a DataGrid row by using the row details functionality.
```

File: controls/datagrid/row-details/row-details-template.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+position: 
+slug: datagrid-row-details-template
+---
+
+# .NET MAUI DataGrid Row Details Template
+
+The Telerik UI for .NET MAUI DataGrid control define a `RowDetailsTemplate` property (type `DataTemplate`). The property is used for displaying row details.
```
Comment (yordan-mitev) [inline controls/datagrid/row-details/row-details-template.md:11]:
```suggestion
The Telerik UI for .NET MAUI DataGrid control defines a `RowDetailsTemplate` property (type `DataTemplate`). The property is used for displaying row details. 
```

File: controls/datagrid/row-details/row-details-template.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+position: 
+slug: datagrid-row-details-template
+---
+
+# .NET MAUI DataGrid Row Details Template
+
+The Telerik UI for .NET MAUI DataGrid control define a `RowDetailsTemplate` property (type `DataTemplate`). The property is used for displaying row details. 
+
+
+## Example
```
Comment (yordan-mitev) [inline controls/datagrid/row-details/row-details-template.md:14]: To be added later? 
You don't need a dedicated section for the example. Adding a sentence that introduces the example is the better approach in such short articles.

File: controls/datagrid/row-details/row-details-overview.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
+---
+title: Overview
+page_title: .NET MAUI DataGrid Documentation - Row Details Overview
+description: Learn how to present additional information about DataGrid row by using row details functionality.
```
Comment (yordan-mitev) [inline controls/datagrid/row-details/row-details-overview.md:4]: Good job on the description!
```suggestion
description: Learn how to present additional information about DataGrid rows by using the row details functionality.
```

File: controls/datagrid/row-details/row-details-overview.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+position: 0
+slug: datagrid-row-details-overview
+---
+
+# .NET MAUI DataGrid Row Details Overview
+
+The Telerik UI for .NET MAUI DataGrid control is capable of presenting additional information with a Row Details functionality. The Row Details is a Data Template defined on the grid- or row-level and is used for displaying data without affecting the dimensions of the row and the cells within it.
```
Comment (yordan-mitev) [inline controls/datagrid/row-details/row-details-overview.md:11]: That's a good introduction!
```suggestion
The Telerik UI for .NET MAUI DataGrid control is capable of presenting additional information through the Row Details functionality. The Row Details is a Data Template defined on the grid- or row-level and is used for displaying data without affecting the dimensions of the row and the cells within it.
```

File: controls/datagrid/row-details/row-details-overview.md [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+![DataGrid Row Details Overview](images/datagrid-row-details.png)
+
+To show the Row Details you can use exposed options:
```
Comment (yordan-mitev) [inline controls/datagrid/row-details/row-details-overview.md:15]:
```suggestion
To show the Row Details, you can use the following exposed options:
```

File: controls/datagrid/row-details/row-details-overview.md [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+- Using the default column `DataGridToggleRowDetailsColumn`&mdash;allows showing and hiding the row details for an item. For additional information you can refer to our DataGrid [Toggle Row Details Column]({% slug datagrid-columns-toggle-column% })
```
Comment (yordan-mitev) [inline controls/datagrid/row-details/row-details-overview.md:17]: - "Use" is already included in the sentence that introduces this list. You don't have to repeat it.
- Avoid "our" and "we".

```suggestion
- The default `DataGridToggleRowDetailsColumn` column &mdash;Allows showing and hiding the row details for an item. For additional information, refer to the DataGrid [Toggle Row Details Column]({% slug datagrid-columns-toggle-column% })
```

File: controls/datagrid/row-details/row-details-overview.md [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+
+- Using `SelectionChanged` event.
```
Comment (yordan-mitev) [inline controls/datagrid/row-details/row-details-overview.md:19]:
```suggestion
- The `SelectionChanged` event.
```

File: controls/datagrid/row-details/row-details-overview.md [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+
+The DataGrid exposes the current properties:
```
Comment (yordan-mitev) [inline controls/datagrid/row-details/row-details-overview.md:21]: Add more detail to this sentence. What is the purpose of these properties? What is the common thing between them?

File: controls/datagrid/row-details/row-details-overview.md [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+
+* `AreNowDetailsFrozen` (type `bool`)&mdash;The property indicates whether the row details keep position during horizontal scroll.
```
Comment (yordan-mitev) [inline controls/datagrid/row-details/row-details-overview.md:23]:
```suggestion
* `AreNowDetailsFrozen` (type `bool`)&mdash;The property indicates whether the row details keep their position during horizontal scroll.
```

**Review by dessyordanova — DISMISSED** [review]
image
Improve the gif file in such a way that all borders of the grid are always visible - make the grid a fixed size an use the same frame size for capturing the gif. Expanding/collapsing wouldn't change the size of the grid but just the visible rows count and you will have clear boundaries of the gif and the grid

File: controls/datagrid/row-details/row-details-overview.md [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+title: Overview
+page_title: .NET MAUI DataGrid Documentation - Row Details Overview
+description: Learn how to present additional information about DataGrid rows by using the row details functionality.
+position: 0
+slug: datagrid-row-details-overview
+---
+
+# .NET MAUI DataGrid Row Details Overview
+
+The Telerik UI for .NET MAUI DataGrid control is capable of presenting additional information through the Row Details functionality. The Row Details is a Data Template defined on the grid- or row-level and is used for displaying data without affecting the dimensions of the row and the cells within it.
```
Comment (yordan-mitev) [inline controls/datagrid/row-details/row-details-overview.md:11]:
```suggestion
The Telerik UI for .NET MAUI DataGrid control is capable of presenting additional information through the Row Details functionality. The Row Details is a Data Template defined on the grid or row level and is used for displaying data without affecting the dimensions of the row and the cells within it.
```

File: controls/datagrid/row-details/row-details-overview.md [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+
+![DataGrid Row Details Overview](images/datagrid-row-details.png)
+
+To show the Row Details, you can use the following exposed options:
+
+- The default `DataGridToggleRowDetailsColumn` column &mdash;Allows showing and hiding the row details for an item. For additional information, refer to the DataGrid [Toggle Row Details Column]({% slug datagrid-columns-toggle-column% })
+
+- The `SelectionChanged` event.
+
+The DataGrid exposes the current properties for row details functionality:
```
Comment (yordan-mitev) [inline controls/datagrid/row-details/row-details-overview.md:21]:
```suggestion
The DataGrid exposes the following properties that control the row details functionality:
```

File: controls/datagrid/row-details/row-details-template.md [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+position: 
+slug: datagrid-row-details-template
+---
+
+# .NET MAUI DataGrid Row Details Template
+
+The Telerik UI for .NET MAUI DataGrid control defines a `RowDetailsTemplate` property (type `DataTemplate`). The property is used for displaying row details. 
+
+
+## Example
```
Comment (yordan-mitev) [inline controls/datagrid/row-details/row-details-template.md:14]:
```suggestion
```

File: controls/datagrid/row-details/row-details-template.md [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+---
+
+# .NET MAUI DataGrid Row Details Template
+
+The Telerik UI for .NET MAUI DataGrid control defines a `RowDetailsTemplate` property (type `DataTemplate`). The property is used for displaying row details. 
+
+
+## Example 
+
+This example provides steps how to define the `RowDetailsTemplate` property in the DataGrid control.
```
Comment (yordan-mitev) [inline controls/datagrid/row-details/row-details-template.md:16]:
```suggestion
The following example shows how to define the `RowDetailsTemplate` property in the DataGrid control.
```

File: controls/datagrid/row-details/row-details-template.md [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+
+3. Define the Business Model:
+
+<snippet id ='datagrid-business-model'/>
+
+4. Define the ItemsSource:
+
+<snippet id ='datagrid-items-source'/>
+
+5. In addition you can define the row details to be expanded:
```
Comment (yordan-mitev) [inline controls/datagrid/row-details/row-details-template.md:34]:
```suggestion
5. In addition, you can define the row details to be expanded:
```

**Review by yordan-mitev — APPROVED** [review]

**Pull request merged**